addons/account/models/sequence_mixin.py

```diff
@@ -72,7 +102,7 @@ def _get_starting_sequence(self):
         return "00000000"
 
     def _get_highest_query(self):
-        return "SELECT {field} FROM {table} {where_string} ORDER BY {field} DESC LIMIT 1 FOR UPDATE"
+        return "SELECT {field} FROM {table} {where_string} ORDER BY sequence_prefix DESC, sequence_number DESC LIMIT 1 FOR UPDATE"
```

Comment:
As confirmed in [this comment](https://github.com/odoo/odoo/pull/49095#discussion_r441542933), this method is not needed anymore. The query can be inlined in `_get_last_sequence` directly.

Also, I'm not convinced of the `ORDER BY sequence_prefix DESC`.
Wouldn't it be better to search for the last used prefix?

Reply:
By doing a sub select with 
```sql
SELECT sequence_prefix FROM account_move {where_string} ORDER BY id DESC
```
And filtering on the found prefix?

I guess it is not a bad idea.

By adding an index with `id`, i get
```sql
saas-13.3-sequence-regex-2-wan=# EXPLAIN (ANALYZE, VERBOSE, BUFFERS) SELECT name FROM account_move
WHERE journal_id = 1 AND name != '/' AND move_type NOT IN ('out_refund', 'in_refund')
AND sequence_prefix = (SELECT sequence_prefix FROM account_move WHERE journal_id = 1 AND name != '/' AND move_type NOT IN ('out_refund', 'in_refund')  ORDER BY id DESC LIMIT 1)
ORDER BY sequence_number DESC
LIMIT 1;
                                                                                   QUERY PLAN                                                                                   
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=8.31..16.33 rows=1 width=36) (actual time=0.055..0.056 rows=1 loops=1)
   Output: account_move.name, account_move.sequence_number
   Buffers: shared hit=4
   InitPlan 1 (returns $0)
     ->  Limit  (cost=0.14..8.17 rows=1 width=36) (actual time=0.032..0.033 rows=1 loops=1)
           Output: account_move_1.sequence_prefix, account_move_1.id
           Buffers: shared hit=2
           ->  Index Scan using account_move_sequence_index2 on public.account_move account_move_1  (cost=0.14..8.17 rows=1 width=36) (actual time=0.031..0.031 rows=1 loops=1)
                 Output: account_move_1.sequence_prefix, account_move_1.id
                 Index Cond: (account_move_1.journal_id = 1)
                 Filter: (((account_move_1.name)::text <> '/'::text) AND ((account_move_1.move_type)::text <> ALL ('{out_refund,in_refund}'::text[])))
                 Buffers: shared hit=2
   ->  Index Scan using account_move_sequence_index on public.account_move  (cost=0.14..8.17 rows=1 width=36) (actual time=0.054..0.054 rows=1 loops=1)
         Output: account_move.name, account_move.sequence_number
         Index Cond: ((account_move.journal_id = 1) AND ((account_move.sequence_prefix)::text = ($0)::text))
         Filter: (((account_move.name)::text <> '/'::text) AND ((account_move.move_type)::text <> ALL ('{out_refund,in_refund}'::text[])))
         Buffers: shared hit=4
 Planning time: 0.623 ms
 Execution time: 0.123 ms
(19 rows)
```